### PR TITLE
feat: HydroCo - Beverage Powders

### PR DESCRIPTION
**Pull request by DisposableCrewmember42**
## About the PR
- Added HydroCo Dairy and Juice crates. They contain beverage powders that, when mixed with water in a 1:1 ratio and stirred, create the advertised beverage.
- Each tin contains 50u of powder, which can create 100u of liquid
- The dairy crate costs 1500 Spesos and contains
  - 2 Milk tins
  - 2 Soy Milk tins
- The juice crate costs 1500 Spesos and contains one tin of each group
  - Citrus
      - Orange
      - Lime
      - Lemon
  - Tropical
      - Pineapple
      - Banana
      - Watermelon
   - Berry
      - Berry
      - Grape
      - Cherry
   - Garden
      - Apple
      - Carrot
      - Tomato
- Both crates also contain two spoons and a note from HydroCo
- The HydroCo (HYD) stock was added to StockTrade
- If powder is eaten straight, it causes coughing. If more than 15u is consumed, there is a chance of vomiting.
- The more acidic juice powders (pineapple, lemon, lime) cause 1 caustic damage per unit consumed
- Added mail containing one random tin of hydroco powder with a note
- Added 2 tins of soy milk powder and 1 tin of orange juice powder to the Sustenance Vendor (commonly found in Perma)

## Why / Balance
Requested in [The Drawing Board](https://discord.com/channels/968983104247185448/1152625672020316200/1306820163622142043) and further discussed in a [dedicated thread](https://discord.com/channels/968983104247185448/1307145855199084644/1307145855199084644).

Regarding balance:
I believe the prices are justified and high enough to discourage buying lots of juice instead of cooperating with Botany, especially since you can't order specific types in bulk.

## Technical details
YML changes mostly in line with everything else in each file/category. For the tins, I created a new base type, since neither the FoodTinBase prototype nor the ReagentPacketBase suited my exact needs.

The texture uses the Sprite color attribute and an overlay to avoid lots of pointless and tiring work, and make it easy to add new types in the future.

## Media

orders
tins
stonks
notice
inspect

https://github.com/user-attachments/assets/2a199100-7938-44ce-92cd-5bbce76dd9da

https://github.com/user-attachments/assets/84ec0289-ac4d-4f5f-a53f-d896859bec2e

## Requirements

- [x] I have tested all added content and changes.
- [x] I have added media to this PR or it does not require an ingame showcase.

## Breaking changes

**Changelog**

:cl: DisposableCrewmember42
- add: NanoTrasen has established business relations with HydroCo! You can now order powdered milk and juice via your Logistics department and bet on the growth of the HYD stock.

### CODE REVIEW

**Comment by IamVelcroboy** [comment]
This is really neat. My main suggestion would be to reduce the contents of the fruit crates to 4 and use rng group [tables](https://docs.spacestation14.com/en/space-station-14/core-tech/entity-tables.html).  
With like: 
- group 1 citrus (Lemon, Lime, Orange) 
- group 2 berry (berry, cherry, grape) 
- group 3 tropical (pineapple, banana, watermelon) 
- group 4 other (apple, carrot)

**Comment by dvir001** [comment]
Please add a crate for energy drinks versions. (GFuel, GamerSupps).

Call it GamerDonkCo / CyberFuel / SyndijuiceSupps etc

**Comment by IamVelcroboy** [comment]
Another suggestion, one or two of these would probably be a good addition to the perma SusVend

**Comment by Avalon-Proto** [comment]
> Another suggestion, one or two of these would probably be a good addition to the perma SusVend

Speaking of that machine, it needs a restock or something. Currently doesn't have one and it'd be nice to be able to restock it so the Perma players have somewhat reliable food besides what they make

**Comment by IamVelcroboy** [comment]
> > Another suggestion, one or two of these would probably be a good addition to the perma SusVend
> 
> Speaking of that machine, it needs a restock or something. Currently doesn't have one and it'd be nice to be able to restock it so the Perma players have somewhat reliable food besides what they make

Definitely out of scope for this PR but it would be an easy PR for anyone to do.

**Comment by DisposableCrewmember42** [comment]
> Please add a crate for energy drinks versions. (GFuel, GamerSupps).
> 
> Call it GamerDonkCo / CyberFuel / SyndijuiceSupps etc

I'll probably PR more variations of beverages at a later point, but I'd consider that out of scope for now. I was thinking of also adding syrups for SpaceUp and stuff like that, but powdered energy drinks would work too.

**Comment by DisposableCrewmember42** [comment]
> This is really neat. My main suggestion would be to reduce the contents of the fruit crates to 4 and use rng group [tables](https://docs.spacestation14.com/en/space-station-14/core-tech/entity-tables.html). With like:
> 
>     * group 1 citrus (Lemon, Lime, Orange)
> 
>     * group 2 berry (berry, cherry, grape)
> 
>     * group 3 tropical (pineapple, banana, watermelon)
> 
>     * group 4 other (apple, carrot)

Makes sense, I'll do that and reduce the price to 1500.

**Comment by DisposableCrewmember42** [comment]
oh god i fucked up the rebase

**Comment by DisposableCrewmember42** [comment]
This doesn't modify C#, label can be removed. Should be the final changes, ready for review!

File: Resources/Prototypes/DeltaV/Catalog/Fills/Crates/food.yml [hunk 1]
```diff
@@ -29,3 +29,119 @@
...
+- type: entity
+  id: CrateHydroCoDairy
+  parent: CratePlastic
+  name: HydroCo Dairy crate
+  description: A shipment of powdered dairy products from HydroCo. Contains powdered milk and soy milk.
+  components:
+    - type: StorageFill
+      contents:
+        - id: PaperCrateHydroCoConsumption
+          amount: 1
```
Comment (IamVelcroboy) [inline Resources/Prototypes/DeltaV/Catalog/Fills/Crates/food.yml]: Amounts of 1 don't need to be specified. They are the default value.

File: Resources/Prototypes/DeltaV/Catalog/Fills/Crates/food.yml [hunk 2]
```diff
@@ -29,3 +29,119 @@
...
+          tableId: FillHydroCoJuiceBerry
+        - !type:NestedSelector
+          tableId: FillHydroCoJuiceTropical
+        - !type:NestedSelector
+          tableId: FillHydroCoJuiceGarden
+        - id: SpoonPlastic
+          amount: !type:ConstantNumberSelector
+            value: 2
+        - id: PaperCrateHydroCoConsumption
+          amount: !type:ConstantNumberSelector
```
Comment (IamVelcroboy) [inline Resources/Prototypes/DeltaV/Catalog/Fills/Crates/food.yml]: Same with this and all the rest down the line. Value of 1 is default.

File: Resources/Prototypes/DeltaV/Catalog/Fills/Crates/food.yml [hunk 3]
```diff
@@ -29,3 +29,119 @@
...
+      amount: !type:ConstantNumberSelector
+        value: 1
+    - id: ReagentTinPowderedJuiceCarrot
+      amount: !type:ConstantNumberSelector
+        value: 1
+    - id: ReagentTinPowderedJuiceTomato
+      amount: !type:ConstantNumberSelector
+        value: 1
+
+- type: entityTable
```
Comment (IamVelcroboy) [inline Resources/Prototypes/DeltaV/Catalog/Fills/Crates/food.yml:109]: This doesn't look to be used anywhere. See comment I left you in Discord.

File: Resources/Prototypes/DeltaV/Catalog/Fills/Crates/food.yml [hunk 4]
```diff
@@ -29,3 +29,119 @@
...
+        - !type:NestedSelector
+          tableId: FillHydroCoJuiceGarden
+        - id: SpoonPlastic
+          amount: !type:ConstantNumberSelector
+            value: 2
+        - id: PaperCrateHydroCoConsumption
+          amount: !type:ConstantNumberSelector
+            value: 1
+
+- type: entityTable
```
Comment (IamVelcroboy) [inline Resources/Prototypes/DeltaV/Catalog/Fills/Crates/food.yml]: Modular tables like this works but they *could* be condensed. See: [comment](https://discord.com/channels/968983104247185448/1307145855199084644/1308253511133171772) in Discord.

**Comment by DisposableCrewmember42** [comment]
i rebased and reworded a commit to retrigger the flaky test and did the thing again, apparently that's not a great idea after merge commit :blunt:

File: Resources/Prototypes/DeltaV/Entities/Objects/Consumable/Drinks/powdered_drinks.yml [hunk 5]
```diff
@@ -0,0 +1,339 @@
+- type: entity
+  id: ReagentTinBase
+  abstract: true
+  parent: ReagentContainerBase
+  components:
```
Comment (MilonPL) [inline Resources/Prototypes/DeltaV/Entities/Objects/Consumable/Drinks/powdered_drinks.yml]: wrong order

File: Resources/Prototypes/DeltaV/Entities/Objects/Misc/paper.yml [hunk 6]
```diff
@@ -30,3 +30,64 @@
   id: PaperFugitiveReport
   name: fugitive report
   description: An arrest warrant for a space fugitive sent from GALPOL.
+
+- type: entity
+  id: PaperCrateHydroCoConsumption
+  categories: [ HideSpawnMenu ]
+  name: HydroCo Notice
+  parent: Paper
```
Comment (MilonPL) [inline Resources/Prototypes/DeltaV/Entities/Objects/Misc/paper.yml]: wrong order

File: Resources/Prototypes/DeltaV/Entities/Objects/Misc/paper.yml [hunk 7]
```diff
@@ -30,3 +30,64 @@
...
+      before consuming.
+
+      Thank you for choosing HydroCo!
+      Enjoy your drinks!
+
+- type: entity
+  id: PaperMailHydroCoTrial
+  categories: [ HideSpawnMenu ]
+  name: HydroCo Trial Message
+  parent: Paper
```
Comment (MilonPL) [inline Resources/Prototypes/DeltaV/Entities/Objects/Misc/paper.yml]: above

**Review by MilonPL — CHANGES_REQUESTED** [review]
https://docs.spacestation14.com/en/general-development/codebase-info/conventions.html#yaml-conventions

**Comment by DisposableCrewmember42** [comment]
not sure why it still shows 1 change requested, i marked everything as resolved

File: Resources/Prototypes/DeltaV/Catalog/Fills/Crates/food.yml [hunk 8]
```diff
@@ -29,3 +29,85 @@
           amount: 1
         - id: FoodBoxDonkpocketDink
           prob: 0.1
+
+- type: entity
+  id: CrateHydroCoDairy
+  parent: CratePlastic
```
Comment (MilonPL) [inline Resources/Prototypes/DeltaV/Catalog/Fills/Crates/food.yml:45]: move parent above id

File: Resources/Prototypes/DeltaV/Catalog/Fills/Crates/food.yml [hunk 9]
```diff
@@ -29,3 +29,85 @@
...
+        - id: SpoonPlastic
+          amount: 2
+        - id: ReagentTinPowderedMilk
+          amount: 2
+        - id: ReagentTinPowderedMilkSoy
+          amount: 2
+
+- type: entity
+  id: CrateHydroCoJuice
+  parent: CratePlastic
```
Comment (MilonPL) [inline Resources/Prototypes/DeltaV/Catalog/Fills/Crates/food.yml:61]: above

File: Resources/Prototypes/DeltaV/Catalog/Fills/Crates/food.yml [hunk 10]
```diff
@@ -29,3 +29,85 @@
...
+        - id: SpoonPlastic
+          amount: !type:ConstantNumberSelector
+            value: 2
+        - id: PaperCrateHydroCoConsumption
+
+- type: entityTable
+  id: FillHydroCoJuiceMix
+  table: !type:AllSelector
+    children:
+    #Citrus
```
Comment (MilonPL) [inline Resources/Prototypes/DeltaV/Catalog/Fills/Crates/food.yml]: comments should have trailing spaces like
`# Comment`

File: Resources/Prototypes/DeltaV/Entities/Objects/Consumable/Drinks/powdered_drinks.yml [hunk 11]
```diff
@@ -0,0 +1,339 @@
+- type: entity
+  parent: ReagentContainerBase
+  id: ReagentTinBase
+  abstract: true
```
Comment (MilonPL) [inline Resources/Prototypes/DeltaV/Entities/Objects/Consumable/Drinks/powdered_drinks.yml]: abstract should be above parent

**Review by MilonPL — CHANGES_REQUESTED** [review]

**Comment by DisposableCrewmember42** [comment]
test failures unrelated

**Comment by DisposableCrewmember42** [comment]
Conflicts resolved

**Review by MilonPL — APPROVED** [review]

**Pull request merged**